how_to_build.md

```diff
@@ -0,0 +1,118 @@
+
+At the time of writing the full build (including all target frameworks) can only run on Windows.
+
+[Partial builds](#building-only-a-subset-of-the-supported-target-frameworks) are supported on Linux (Ubuntu 18.04). This might also run on macOS, but hasn't been tested.
+
+## Prerequisites
+
+The build requires that a few pieces of software be installed on the host computer. We're somewhat aggressive about adoptiong new language features and the like, so rather than specifying exactly which versions are required, we'll tend toward "latest" or "at least" forms of guidance. If it seems you have an incompatible version of the software, prefer to upgrade rather than downgrade.
+
+### On Windows
+
+Ensure that the following are installed:
+
+1. a recent version of Visual Studio 2019 (currently this means 16.3 or later) or the Build Tools for Visual Studio 2019
+
+2. The .NET Core 1.0 runtime
+
+3. The .NET Core 2.1 runtime
+
+4. The .NET Framework 4.6.1 or higher
+
+5. The targeting packs for .NET Framework 4.0 and 4.5
+
+6. A recent version of the .NET Core 3.0 SDK (currently this means 3.0.100 or later)
+
+You might not need everything to run a [partial build](#building-only-a-subset-of-the-supported-target-frameworks).
+
+### On Linux
+
+The default [build profile](#building-only-a-subset-of-the-supported-target-frameworks) on Linux is `netcore2+`, which runs the tests on .NET Core 2.1 and 3.0. The .NET Framework isn't supported on Linux, and .NET Core 1.0 is no longer supported, which makes it difficult (if not impossible) to install on Linux.
+
+Ensure the following are installed:
+
+1. The .NET Core 2.1 runtime
+
+2. A recent version of the .NET Core 3.0 SDK (currently this means 3.0.100 or later)
+
+## Building
+
+Using a command prompt, navigate to your clone root folder and execute:
+
+- `build.cmd` on Windows
+- `./build.sh` on Linux
+
+This executes the default build targets to produce .NET Standard and/or .NET Framework artifacts, depending on the selected [build profile](#building-only-a-subset-of-the-supported-target-frameworks).
+
+After the build has completed, the build artifacts will be located in `artifacts`.
+
+## Extras
+
+### Running specific build tasks
+
+`build.cmd` wraps a [Bullseye](https://github.com/adamralph/bullseye) targets project, so you can use all the usual command line arguments that you would use with Bullseye, e.g.:
+
+* View the full list of build targets:
+
+    `build.cmd -T`
+
+* Run a specific target:
+
+    `build.cmd unit`
+
+* Run multiple targets:
+
+    `build.cmd unit pack`
+
+* Run a target without running its dependencies (might fail if the dependencies
+  haven't been previously built):
+
+    `build.cmd -s pack`
+
+* View the full list of options:
+
+    `build.cmd -?`
+
+(On Linux, just replace `build.cmd` with `./build.sh`)
+
+### Building only a subset of the supported target frameworks

```

Comment:
It is worth keeping the profiles for the analyzers? They build and test pretty fast (for now).

Reply:
Probably not. I had not included this in my attempt to move the analyzers.